src/dialog-user/services/dialogData.ts

```diff
@@ -17,26 +17,37 @@ export default class DialogDataService {
 
     const sortableFieldTypes = ['DialogFieldDropDownList', 'DialogFieldRadioButton'];
     if (_.includes(sortableFieldTypes, field.type)) {
-      const dropDownValues = [];
-      for (let option of field.values) {
-        if (option[0] === String(field.default_value)) {
-          field.selected = option;
-        }
-        const value = ((field.data_type === 'integer' && option[0] !== null) ? parseInt(option[0], 10) : option[0]);
-        const description = (!Number.isInteger(option[1]) ? option[1] : parseInt(option[1], 10));
-        dropDownValues.push([value, description]);
-      }
-      field.values = dropDownValues;
-      if (data.options.sort_by !== 'none') {
-        field.values = this.updateFieldSortOrder(field);
-      }
+      field.values = this.setupSortableValues(field);
     }
 
     field.default_value = this.setDefaultValue(field);
 
     return field;
   }
 
+  // converts values to the right data_type, and order
+  public setupSortableValues({
+    data_type,
+    options,
+    values,
+  }) {
+    let dropDownValues = values.map((option) => {
+      const value = this.convertDropdownValue(option[0], data_type);
+      const description = (!Number.isInteger(option[1]) ? option[1] : parseInt(option[1], 10));
```

Comment:
I don't think we need to convert the description, do we? The `default_value` I believe should always be looking at the `values` and never the description, right? Also, even if we did want to convert the description, could we not just use the new `convertDropdownValue` function?

Reply:
This is not new, we were already converting the description - that comes from https://github.com/ManageIQ/ui-components/pull/265.

I *think* this was meant to fix sorting by description when you have all numeric labels, so I just left it as is, without any conversion.

If we use `convertDropdownValue` here, we need to specify that `data_type` also applies to description, not only label, that doesn't sound quite right.

Reply:
@himdel Ah gotcha, alright, no problem.